templates/jail.local.j2

```diff
@@ -1,3 +1,9 @@
+{% if ansible_distribution_major_version | int >= 12 %}
+[DEFAULT]
+backend = systemd
```

Comment:
It would probably not work for everything, since for example sshd jail has its own definition that overwrites the default:
https://github.com/fail2ban/fail2ban/blob/44fa2959e7c8ee010138250b3dafcfebc57dbce8/config/jail.conf#L282
So `paths-debian.conf` may be adjusted instead (or together).
From other point of view for other jails which remains file-related that may be too much (because the user would even not notice that some jails doesn't match anymore at all.

Reply:
I created my own patch that is now running successfully on my machines, based on this PR and the comments in the review: https://github.com/maveonair/ansible-role-security/commit/376a84f4a3fc1d0767d864d30ad415e032b84cde 

My suggestion would also be that we only set the backend for SSH in `jail.local` if Debian is recognized and version >= 12 is running.

Reply:
Now is the `backend = systemd` configuration in `jail.local`, see https://github.com/geerlingguy/ansible-role-security/commit/0b9a774f875ff3c175fc4f16a6d2b5b479d17e45 templates/jail.local.j2

I did tests, looks good for me

```log
pi@kube2:~ $ sudo journalctl -u fail2ban.service -f

Dec 18 18:53:48 kube2 fail2ban[339242]:   maxLines: 1
Dec 18 18:53:48 kube2 fail2ban[339242]: [sshd] Added journal match for: '_SYSTEMD_UNIT=sshd.service + _COMM=sshd'
Dec 18 18:53:48 kube2 fail2ban[339242]:   maxRetry: 5
Dec 18 18:53:48 kube2 fail2ban[339242]:   findtime: 600
Dec 18 18:53:48 kube2 fail2ban[339242]:   banTime: 600
Dec 18 18:53:48 kube2 fail2ban[339242]:   encoding: UTF-8
Dec 18 18:53:48 kube2 fail2ban[339242]: Jail 'sshd' reloaded
Dec 18 18:53:48 kube2 fail2ban[339242]: Reload finished.
Dec 18 18:53:48 kube2 fail2ban-client[3785923]: OK
Dec 18 18:53:48 kube2 systemd[1]: Reloaded fail2ban.service - Fail2Ban Service.


Dec 18 18:59:04 kube2 fail2ban[339242]: [sshd] Found 192.168.1.3 - 2023-12-18 18:59:03
Dec 18 18:59:04 kube2 fail2ban[339242]: [sshd] Found 192.168.1.3 - 2023-12-18 18:59:04
```